src/odemis/driver/smartpod.py

```diff
@@ -0,0 +1,840 @@
+
+import os
+import logging
+import time
+from ctypes import *
+import copy
+import threading
+
+from odemis import model
+from odemis.util import driver
+from odemis.model import HwError, CancellableFuture, CancellableThreadPoolExecutor, isasync
+from operator import pos
+
+
+def add_coord(pos1, pos2):
+    """
+    Adds two coordinate dictionaries together and returns a new coordinate dictionary.
+
+    NOTE: pos1 and pos2 MUST contain the same axes values
+
+    pos1: dict (axis name str) -> (float)
+    pos2: dict (axis name str) -> (float)
+    Returns ret
+        dict (axis name str) -> (float)
+    """
+    if pos1.keys() != pos2.keys():
+        raise ValueError("The coordinates added must have the same axes")
+
+    ret = pos1.copy()
+    for an, v in pos2.items():
+        ret[an] += v
+
+    return ret
+
+
+class Pose(Structure):
+    """
+    SmartPod Pose Structure (C Struct used by DLL)
+    """
+    _fields_ = [
+        ("positionX", c_double),
+        ("positionY", c_double),
+        ("positionZ", c_double),
+        ("rotationX", c_double),
+        ("rotationY", c_double),
+        ("rotationZ", c_double),
+        ]
+    
+    def __add__(self, o):
+        pose = Pose()
+        pose.positionX = self.positionX + o.positionX
+        pose.positionY = self.positionY + o.positionY
+        pose.positionZ = self.positionZ + o.positionZ
+        pose.rotationX = self.rotationX + o.rotationX
+        pose.rotationY = self.rotationY + o.rotationY
+        pose.rotationZ = self.rotationZ + o.rotationZ
+        return pose
+
+    def __sub__(self, o):
+        pose = Pose()
+        pose.positionX = self.positionX - o.positionX
+        pose.positionY = self.positionY - o.positionY
+        pose.positionZ = self.positionZ - o.positionZ
+        pose.rotationX = self.rotationX - o.rotationX
+        pose.rotationY = self.rotationY - o.rotationY
+        pose.rotationZ = self.rotationZ - o.rotationZ
+        return pose
+
+
+def pose_to_dict(pose):
+    """
+    Convert a Pose (C structure) to a coordinate dictionary (str) -> (double)
+        pose: a Pose (C struct)
+    returns: Coordinate dictionary (str) -> (double) of axis name to value
+    """
+    pos = {}
+    pos['x'] = pose.positionX
+    pos['y'] = pose.positionY
+    pos['z'] = pose.positionZ
+    pos['theta_x'] = pose.rotationX
+    pos['theta_y'] = pose.rotationY
+    pos['theta_z'] = pose.rotationZ
+    return pos
+
+def dict_to_pose(pos):
+    """
+    Convert a coordinate dictionary (str) -> (double) to a Pose C struct
+        pos: Coordinate dictionary (str) -> (double) of axis name to value
+    returns: a Pose (C struct)
+    raises ValueError if an unsupported axis name is input
+    """
+    pose = Pose()
+    for an, v in pos.items():
+        if an == "x":
+            pose.positionX = v
+        elif an == "y":
+            pose.positionY = v
+        elif an == "z":
+            pose.positionZ = v
+        elif an == "theta_x":
```

Comment:
The convention in Odemis is to call the rotation axes "rx/ry/rz". So I'd suggest to follow the same naming convention here. Alternatively, the user would be able to override the names via the "axes" parameter, but let's not support that yet.